pkg/pgmodel/ingestor/duplicates.go

```diff
@@ -0,0 +1,31 @@
+package ingestor
+
+import (
+	"github.com/timescale/promscale/pkg/clockcache"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/metrics"
+)
+
+const maxDuplicateStoreSize = 10000
+
+var duplicatesStore *clockcache.Cache
+
+func init() {
+	duplicatesStore = clockcache.WithMax(maxDuplicateStoreSize)
+}
+
+func registerDuplicates(tableName string, count int64) {
+	netDuplicates := count
+	previousCountInf, found := duplicatesStore.Get(tableName)
+	if found {
+		previousCount := previousCountInf.(int64)
+		netDuplicates += previousCount
+	} else {
+		// We log the duplicates once per table name. Detailed logging affects performance and hence must be avoided.
+		// As such, details related to counts, etc, can be visualised as prometheus metrics.
+		log.Warn("msg", "duplicate data in sample", "table", tableName, "duplicate_count", count)
+	}
+	metrics.DuplicateSamples.Add(float64(count))
+	metrics.DuplicateWrites.Inc()
+	duplicatesStore.Insert(tableName, netDuplicates)
```

Comment:
we never actually use netDeplicates and Insert is a much more expensive operation than Get, so I'd prefer to store a constant but avoid Inserts in all cases other than cache miss.

Reply:
> Insert is a much more expensive operation than Get

Insert internally first call the gets (via checking `self.elements[]`). So, isn't it the same?

Reply:
The recent commit pushed is in line with what you expect.

Reply:
Also, I feel a bug in `insert()` of `cache.go`. The insert expects key and value. That means, it should store the value. If someone inserts a pair with the key being already in the cache, but with an updated value, the updated value will never be accepted as, if the key is present, the `insert()` is just returning the previous value. But in actuality, it should return only if the key is present and the incoming value is the same as the present value in the cache.

Reply:
@cevian